lib/filter/filter-call.c

```diff
@@ -62,6 +63,13 @@ filter_call_init(FilterExprNode *s, GlobalConfig *cfg)
   FilterCall *self = (FilterCall *) s;
   LogExprNode *rule;
 
+  if (self->visited)
+    {
+      msg_error("Loop detected in filter rule", evt_tag_str("rule", self->rule));
+      return FALSE;
+    }
+  self->visited = TRUE;
+
   /* skip initialize if filter_call_init already called. */
   if (self->filter_expr)
     return TRUE;
```

Comment:
Is it possible that `self->filter_expr` is not `NULL`  when we check the `visited` flag?